pkg/indicator/supertrendPivot.go

```diff
@@ -0,0 +1,217 @@
+package indicator
+
+import (
+	"math"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/c9s/bbgo/pkg/datatype/floats"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+// based on "Pivot Point Supertrend by LonesomeTheBlue" from tradingview
+
+var logpst = logrus.WithField("indicator", "pivotSupertrend")
+
+//go:generate callbackgen -type PivotSupertrend
+type PivotSupertrend struct {
+	types.SeriesBase
+	types.IntervalWindow
+	ATRMultiplier float64 `json:"atrMultiplier"`
+	PivotWindow   int     `json:"pivotWindow"`
+
+	AverageTrueRange *ATR // Value must be set when initialized in strategy
+
+	PivotLow  *PivotLow  // Value must be set when initialized in strategy
+	PivotHigh *PivotHigh // Value must be set when initialized in strategy
```

Comment:
I don't see the reason why the pivots should be here with supertrend instead of standalone indicator

Reply:
The Update function is not the same for supertrend and supertrendPivot. There is not a big difference, but I've tried to keep it similar to the tradingview script (Pivot Point Supertrend by LonesomeTheBlue).

Reply:
what i mean is that in this supertrendPivot, supertrend and pivot high/low seems irrelevant. Maybe pivot high/low can be a standalone indicator? for the supertrend part,  if it's better than the original supertrend indicator, you can improve the supertrend indicator. Or is there any reason to put both pivot and supertrend into an indicator?

Reply:
supertrendPivot calculates the support and resistance lines based on the pivot high/low (not closePrice). SupertrendPivot is not necessarily better than the original, but the result is a bit different as the original supertrend is calculated based on close price. When using pivot high/low instead of close price, it typically gives a bit fewer signals. Also, supertrendPivot is weighting the src. Maybe I'm overlooking something, but I can't see how a standalone pivot high/low indicator and supertrend could generate the same output?

Reply:
oh, i see it now. Sorry that i missed the inc.src part. It uses last pivot and last src to calculate the new src, right?